core/src/main/java/org/elasticsearch/bootstrap/BootstrapCheck.java

```diff
@@ -585,4 +586,32 @@ public String errorMessage() {
 
     }
 
+    static class DefaultClusterNameCheck implements Check {
+
+        private final ClusterName clusterName;
+
+        DefaultClusterNameCheck(final ClusterName clusterName) {
+            this.clusterName = clusterName;
+        }
+
+        @Override
+        public boolean check() {
+            return clusterName.equals(ClusterName.DEFAULT);
```

Comment:
I think that'd be a little dangerous. I mean, not as dangerous now that we only do unicast discovery, but still a little silly. It wouldn't prevent misconfigured nodes from joining the cluster.